app/models/master_file.rb

```diff
@@ -559,32 +559,35 @@ def find_frame_source(options={})
     options[:offset] ||= 2000
 
     source = FileLocator.new(working_file_path&.first || file_location)
-    options[:master] = true
-    if source.source.nil? or (source.uri.scheme == 's3' and not source.exist?)
+    options[:non_temp_file] = true
+    if source.source.blank? or (source.uri.scheme == 's3' and not source.exist?)
       source = FileLocator.new(self.derivatives.where(quality_ssi: 'high').first.absolute_location)
-      options[:master] = false
+      options[:non_temp_file] = true
```

Comment:
I think this is the key change for keeping derivative files from getting deleted by the unlinking on line 639.  I changed the key to better reflect the expectations around this.